File: novel-collection-backend/src/main/java/pj/models/Novel.java
package pj.models;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.LazyToOne;
import org.hibernate.annotations.LazyToOneOption;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Data
@NoArgsConstructor
@Table(indexes = {@Index(name = "nv_slug", columnList = "slug", unique = true), @Index(name = "nv_author", columnList = "author"), @Index(name = "nv_type", columnList = "type"), @Index(name = "nv_status", columnList = "status")})
public class Novel {
    
    @Id  @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    
    private String slug;
    private String title;
    
    @Column(columnDefinition="TEXT")
    private String description;
    
    private String author;
    private String thumbnailURL;
    
    private String type; // Chinese, Korean....
    private String status; // Completed, Ongoing, Canceled, On Hold
    
    private Integer lastReleasedChapter;
    private String lastReleasedDate;
    
    private String source;
    private Float rating;
    
    // You can specify fetch = FetchType.EAGER in @OneToMany. This will fetch all 'Genre' entities (oneToMany) related to a novel when you do a find on a novel id.
   
    @ManyToMany(cascade = CascadeType.MERGE, fetch = FetchType.LAZY)
    private Set<Genre> genres;
    
    @Transient
	Boolean added;
}

File: novel-collection-frontend/src/app/model/novel.ts

export class Novel {
  id: number;
  slug: string;
  title: string;
  description: string;
  author: string;
  thumbnailURL: string;

  type: string; // Chinese, Korean....
  status: string; // Completed, Ongoing, Canceled, On Hold

  lastReleasedChapter: number;
  lastReleasedDate: string;

  source: string;
  rating: number;

  genres: Genre[];
  added: boolean;
  readmore: boolean = false;
}

export class Genre {
  id: number;
  genre: string;
}

export class Chapter {
  compositeKey: ChapterKey;
	title : string;
	content : string;
}

export class ChapterKey {
  novel: Novel
  number: number
}